unstructured/ingest/v2/README2.md

```diff
@@ -0,0 +1,311 @@
+        
+        destination_connection_config=ChromaConnectionConfig(
+            access_config=ChromaAccessConfig(settings=None, headers=None),
+            host="localhost",
+            port=8000,
+            collection_name=f"test-collection-{random.randint(1000,9999)}",
+        ),
+        stager_config=ChromaUploadStagerConfig(),
+        uploader_config=ChromaUploaderConfig(batch_size=10),
+    ).run()
+
+```
+
+Notice how the top part looks similar to the local connector running file. But now we are adding a chunker and an embedder. And the destination connection is for the Chroma uploader. Also note that there is a stager_config. This is where we prepare the document/artifact in a custom way before running the Uploader.
+
+Let's run it.
+
+
+
+* Make sure your python environment is set up and then run `pip install "unstructured[chroma]"`
+
+* In a separate terminal (with chroma installed) run
+`./scripts/chroma-test-helpers/create-and-check-chroma.sh chroma-db-file`
+the service should now be running on port 8000
+
+* Run your example file: `python chroma.py`
+* You can examine the resulting sqlite database (`chroma.sqlite3`) in the `chroma-db-file` directory if you want to see the results.
+
+
+Let's look at the python file in the Unstructured repo
+
+https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/ingest/v2/processes/connectors/chroma.py
+
+* ChromaAccessConfig - Needed for connecting to Chroma. Usually sensitive attributes that will be hidden.
+
+* ChromaConnectionConfig - Non sensitive attributes. `collection_name` does not have a default value. `access_config` imports the ChromaAccessConfig and hides the values via `enhanced_field(sensitive=True)`
+
+* ChromaUploadStagerConfig - The Stager config. Didn't need anything for Chroma.
+
+* ChromaUploadStager - The conform_dict is the critical method here. It takes the file we get from the Embedder step and prepares it for upload to the Chroma database. But it does not upload it. It saves the file to the `upload_stage` directory. The file type can be whatever makes sense for the Uploader phase.
+
+* ChromaUploaderConfig - Attributes that are necessary for the upload stage specifically. The ChromaUploader will be upserting artifacts in batches.
+
+* ChromaUploader - Connects to the Client. And uploads artifacts. Note that it does the minimum amount of processing possible to the artifacts before uploading. The Stager phase is responsible for preparing artifacts. Chroma wants artifacts in a dictionary of lists so we do have to create that in the Uploader since there is not a practical way to store that in a file.
+
+* chroma_destination_entry - Registers the Chroma destination connector with the pipeline. (!!! LINK `unstructured/ingest/v2/processes/connectors/__init__.py`)
+
+Note that the `chroma.py` file imports the official Chroma python package when it *creates* the client and not at the top of the file. This is so that BLABLABLA
```

Comment:
@rbiseck3 can you leave a quick comment on the BLABLABLA

Reply:
Note that the `chroma.py` file imports the official Chroma python package when it *creates* the client and not at the top of the file. This allows the classes to be instantiated without error and only cause a runtime error due to missing imports.